File: src/main/java/me/centralhardware/znatoki/telegram/statistic/telegram/CommandHandler/studentCommand/UserInfoCommand.java
package me.centralhardware.znatoki.telegram.statistic.telegram.CommandHandler.studentCommand;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import me.centralhardware.znatoki.telegram.statistic.entity.postgres.Client;
import me.centralhardware.znatoki.telegram.statistic.entity.postgres.Role;
import me.centralhardware.znatoki.telegram.statistic.i18n.MessageConstant;
import me.centralhardware.znatoki.telegram.statistic.mapper.postgres.ServiceMapper;
import me.centralhardware.znatoki.telegram.statistic.mapper.postgres.ServicesMapper;
import me.centralhardware.znatoki.telegram.statistic.mapper.postgres.UserMapper;
import me.centralhardware.znatoki.telegram.statistic.service.ClientService;
import me.centralhardware.znatoki.telegram.statistic.telegram.TelegramUtil;
import me.centralhardware.znatoki.telegram.statistic.telegram.CommandHandler.CommandHandler;
import org.springframework.stereotype.Component;
import org.telegram.telegrambots.meta.api.objects.Message;

import java.util.Optional;

/**
 * get user info by id
 * param: id of pupil
 * output format: "pupil toString"
 * input format: "/command pupil-id"
 * access level: read
 */
@Slf4j
@Component
@RequiredArgsConstructor
public class UserInfoCommand extends CommandHandler {

    private final ClientService clientService;
    private final TelegramUtil telegramUtils;
    private final UserMapper userMapper;
    private final ServiceMapper serviceMapper;
    private final ServicesMapper servicesMapper;

    @Override
    public void handle(Message message) {
        var arguments = message.getText().replace("/i ", "");
        clientService.findById(Integer.valueOf(arguments)).ifPresentOrElse(
                client -> {
                    var orgId = userMapper.getById(message.getFrom().getId()).getOrganizationId();
                    if (!client.getOrganizationId().equals(orgId)){
                        sender.sendText("Доступ запрещен", message.getFrom());
                        return;
                    }

                    sender.sendMessageWithMarkdown(client.getInfo(serviceMapper.getServicesForCLient(client.getId()).stream().map(servicesMapper::getNameById).toList()), message.getFrom());
                },
                () -> sender.sendMessageFromResource(MessageConstant.PUPIL_NOT_FOUND, message.getFrom())
        );
    }

    @Override
    public boolean isAcceptable(String data) {
        return data.startsWith("/i");
    }

    @Override
    public Role getRequiredRole() {
        return Role.READ;
    }

}

File: src/main/kotlin/me/centralhardware/znatoki/telegram/statistic/telegram/commandHandler/studentCommand/UserInfoCommand.kt
package me.centralhardware.znatoki.telegram.statistic.telegram.commandHandler.studentCommand

import dev.inmo.tgbotapi.extensions.api.send.sendMessage
import dev.inmo.tgbotapi.extensions.behaviour_builder.BehaviourContext
import dev.inmo.tgbotapi.extensions.behaviour_builder.triggers_handling.onCommandWithArgs
import me.centralhardware.znatoki.telegram.statistic.entity.StudentId
import me.centralhardware.znatoki.telegram.statistic.entity.getInfo
import me.centralhardware.znatoki.telegram.statistic.entity.toStudentId
import me.centralhardware.znatoki.telegram.statistic.mapper.StudentMapper
import me.centralhardware.znatoki.telegram.statistic.mapper.LessonMapper
import me.centralhardware.znatoki.telegram.statistic.mapper.SubjectMapper

fun BehaviourContext.userInfoCommand() = onCommandWithArgs("i") { message, args ->
    StudentMapper.findById(args.first().toInt().toStudentId())?.let { client ->
        sendMessage(
            message.chat,
            client.getInfo(
                LessonMapper.getSubjectIdsForStudent(client.id!!)
                    .mapNotNull { SubjectMapper.getNameById(it) }
                    .toList()
            ),
        )
    } ?: sendMessage(message.chat, "Ученик не найден")
}
